gcc/brig/brigfrontend/brig-copy-move-inst-handler.cc

```diff
@@ -53,12 +53,12 @@ brig_copy_move_inst_handler::operator () (const BrigBase *base)
 
   tree input = build_tree_operand_from_brig (brig_inst, source_type, 1);
   tree output = build_tree_operand_from_brig (brig_inst, dest_type, 0);
+
   if (brig_inst->opcode == BRIG_OPCODE_COMBINE)
     {
       /* For combine, a simple reinterpret cast from the array constructor
 	 works.  */
-
-      tree casted = build_reinterpret_cast (dest_type, input);
+      tree casted = build_reinterpret_cast (TREE_TYPE (output), input);
```

Comment:
Why does dest_type and TREE_TYPE (output) differ here? Sounds like a problem in gccbrig_tree_type_for_hsa_type()?

Reply:
dest_type is tree type for combine_vX_bY_bZ so its always unsigned int type. output tree is 'register' variable which can be any type in this patch.